File: demo_app/app/src/main/java/com/bandyer/demo_android_sdk/adapter_items/NoUserSelectedItem.java
package com.bandyer.demo_android_sdk.adapter_items;

import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;

import com.bandyer.demo_android_sdk.R;
import com.bandyer.demo_android_sdk.utils.Utils;
import com.mikepenz.fastadapter.items.AbstractItem;
import java.util.List;

import butterknife.BindView;

/**
 * A simple RecyclerView item used to display no selections on users' list.
 */
public class NoUserSelectedItem extends AbstractItem<NoUserSelectedItem, NoUserSelectedItem.ViewHolder> {

    public static long NO_USER_SELECTED_ITEM_IDENTIFIER = Long.MAX_VALUE;

    public NoUserSelectedItem() {
        super();
        withIdentifier(NO_USER_SELECTED_ITEM_IDENTIFIER);
    }

    @Override
    public int getType() {
        return R.id.no_user_selecte_item;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.no_users_selected_item;
    }

    @NonNull
    @Override
    public NoUserSelectedItem.ViewHolder getViewHolder(@NonNull View v) {
        return new NoUserSelectedItem.ViewHolder(v);
    }

    protected static class ViewHolder extends ButterKnifeViewHolder<NoUserSelectedItem> {

        @BindView(R.id.no_user_selecte_item)
        TextView noUserselectedItem;

        ViewHolder(View view) {
            super(view);
            view.setFocusable(false);
            view.setFocusableInTouchMode(false);
            view.setClickable(false);
        }

        @Override
        public void bindView(@NonNull NoUserSelectedItem item, @NonNull List<Object> payloads) {
            noUserselectedItem.setText(noUserselectedItem.getContext().getString(R.string.no_users_selected));
        }

        @Override
        public void unbindView(@NonNull NoUserSelectedItem item) { }
    }
}

File: app_utilities/src/main/java/com/bandyer/app_utilities/adapter_items/NoUserSelectedItem.kt
/*
 * Copyright (C) 2020 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.app_utilities.adapter_items

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bandyer.app_utilities.R
import com.mikepenz.fastadapter.items.AbstractItem
import kotlinx.android.synthetic.main.no_users_selected_item_layout.view.*

/**
 * A simple RecyclerView item used to display no selections on users' list.
 */
class NoUserSelectedItem : AbstractItem<NoUserSelectedItem.ViewHolder>() {

    override var identifier: Long =  NO_USER_SELECTED_ITEM_IDENTIFIER
    override val type: Int = R.id.no_user_selecte_item
    override val layoutRes: Int = R.layout.no_users_selected_item_layout
    override fun getViewHolder(v: View): ViewHolder = ViewHolder(v)

    override fun bindView(holder: ViewHolder, payloads: List<Any>) {
        super.bindView(holder, payloads)
        holder.containerView.no_user_selecte_itemView.text = holder.containerView.context.getString(R.string.no_users_selected)
    }

    class ViewHolder(val containerView: View) : RecyclerView.ViewHolder(containerView) {

        init {
            containerView.isFocusable = false
            containerView.isFocusableInTouchMode = false
            containerView.isClickable = false
        }
    }

    companion object {
        @JvmField
        var NO_USER_SELECTED_ITEM_IDENTIFIER = R.id.no_user_selecte_item.toLong()
    }
}